### DISCUSSION TITLE
New on x/: Increased Config Flexibility, New Capabilities, Tooling and Bug Fixes

### DISCUSSION THREAD

**Discussion by fortuna** [OutlineFoundation/outline-sdk#239] [Announcements]
## Key Improvements:

- **Effortlessly Extend Configurations**: Add your own custom transports without the need to modify core code.
- **Transport Config Versatility**: Seamlessly combine packet and stream dialers for previously impossible scenarios. Think UDP over SOCKS5, DNS-over-UDP for stream dialers, Encrypted DNS for Packet Dialers, UDP over Websockets, and more!
- **New Configurable Transports**: Enjoy DNS-over-HTTPS, regular DNS on port 53 (TCP+UDP), WebSockets and SOCKS5 with authentication support.
- **Websocket Reverse Proxy**: A new tool to map websockets to Stream or Packet endpoints.
- **Desktop Circumvention Made Easy**: Introducing the `sysproxy` package to effortlessly set system-wide proxies and create desktop circumvention applications.

## Important Bug Fixes (Mobile Proxy):

Ensure a smooth experience by upgrading your mobileproxy to benefit from crucial connection management fixes.

## Relevant PRs:
- Custom Transports: https://github.com/Jigsaw-Code/outline-sdk/pull/213
- Configuration Revamp: https://github.com/Jigsaw-Code/outline-sdk/pull/230
- SOCKS5 Authentication: https://github.com/Jigsaw-Code/outline-sdk/pull/204
- DNS-over-HTTPS: https://github.com/Jigsaw-Code/outline-sdk/pull/227
- Websocket Reverse Proxy: https://github.com/Jigsaw-Code/outline-sdk/pull/202
- System Proxy Setup: https://github.com/Jigsaw-Code/outline-sdk/pull/190
- Mobile Proxy Fixes: https://github.com/Jigsaw-Code/outline-sdk/pull/206 and https://github.com/Jigsaw-Code/outline-sdk/pull/212

**Comment by 62w71st** [comment]
@fortuna @jyyi1 is websocket supported in outline clients yet? Are there plans or timelines for that support to rollout? I'm interested in rolling websocket to a large user base.

**Reply by jyyi1** [reply]
WebSocket support is not currently available in Outline clients. However, we are actively exploring adding it in this year.
